rootpy/stats/dataset.py

```diff
@@ -13,9 +13,31 @@
     'DataSet',
 ]
 
-
 class DataSet(NamedObject, QROOT.RooDataSet):
     _ROOT = QROOT.RooDataSet
+    class Entry(object):
+        def __init__(self, idx, dataset):
+            self.idx_ = idx
+            self.dataset_ = dataset
+            
+        @property
+        def leaves(self):
+            return asrootpy(self.dataset_.get(self.idx_))
+        
+        @property
+        def weight(self):
+            self.dataset_.get(self.idx_) #set current event
```

Comment:
Not sure about that. Say that you have the following:

```
dset = tfile.dataset #get some dataset
fisrt_entry = dset[0]
second_entry = dset[1]
print fisrt_entry.weight
```
I am not sure if this will print the weight of first entry or (more likely) of the second.

Another option would be to store the fields and the weight internally in the Entry class, but this risks precluding the option of assignment of values in future developments